addons/base_automation/models/base_automation.py

```diff
@@ -64,6 +65,17 @@
 ]
 
 
+def get_domain_fields(domain):
```

Comment:
```py
def get_fields_in_domain(domain):
   if isinstance(domain, str):
      ast_list = None 
      for node in ast.walk(ast.parse(domain)):
         if isinstance(node, ast.List):
            ast_list = node
            break
      if ast_list:
         for node in ast.iter_child_nodes(ast_list):
            if isinstance(node, ast.Tuple) and len(node.elts) == 3 and isinstance(node.elts[0], ast.Constant) and isinstance(node.elts[1], ast.Constant):
                yield node.elts[0].value.split(".")[0]
```

to avoid going deeper into the expression and including tuples that are not leafs